### DISCUSSION TITLE
cobbler 2.8.4 install ubuntu16.04&18.04 "d-i grub-installer/bootdev string /dev/sda" does not work

### DISCUSSION THREAD

**Discussion by yogi401** [cobbler/cobbler#2581] [Q&A]
I'm setting up an automatic install of Ubuntu 16.04&18.04, using preseed. the preseed file like this :
```
d-i partman-auto/method string regular 
d-i partman-auto/disk string /dev/sda
d-i partman-auto/choose_recipe select atomic
d-i partman-lvm/device_remove_lvm boolean true
d-i partman-md/device_remove_md boolean true

d-i partman/confirm_write_new_label boolean true
d-i partman-partitioning/confirm_write_new_label boolean true
d-i partman/confirm_nooverwrite boolean true
d-i partman/confirm boolean true
d-i partman/choose_partition select Finish partitioning and write changes to disk

d-i grub-installer/only_debian boolean true
d-i grub-installer/with_other_os boolean true
d-i grub-installer/bootdev string /dev/sda
```
During the system installation process, I still prompted  to fill in the location of the grub installation. it seems that  "d-i grub-installer/bootdev string /dev/sda " not worked.
but when i use the same preseed file to install ubuntu 14.04 ,the Installation can be completed normally.
what is the difference between these two installation? how can i do to complate 16 and 18 installation?
 the screenshot:
<img width="608" alt="WX20190905-110340@2x" src="https://user-images.githubusercontent.com/1264604/64309074-fd835a00-cfcd-11e9-9440-fa75cea151c6.png">

**Comment by yogi401** [comment]
after my test, i put "d-i grub-installer/bootdev string /dev/sda" into kernel option, just like "netcfg/choose_interface=eth0", the installation completed, but i don't know why  
```
LABEL ubuntu-16-x86_64
        kernel /images/ubuntu-16-x86_64/linux
        MENU LABEL ubuntu-16-x86_64
        append initrd=/images/ubuntu-16-x86_64/initrd.gz ksdevice=bootif lang=  netcfg/get_nameservers=8.8.8.8 text netcfg/choose_interface=eth0 grub-installer/bootdev=/dev/sda  auto-install/enable=true priority=critical url=http://172.16.10.66/cblr/svc/op/ks/profile/ubuntu-16-x86_64 hostname=ubuntu-16-x8664 domain=local.lan suite=xenial
```

**Comment by SchoolGuy** [comment]
@yogi401 Thanks for sharing I will try to think of the why on the weekend!

**Comment by SchoolGuy** [comment]
@yogi401 I didn't find any hard evidence and havn't had the time to try it but it seems to me that

```
d-i grub-installer/only_debian boolean true
d-i grub-installer/with_other_os boolean true
```
or mutually exclusive to one another. Setting both would be wrong but I am not sure and will confirm for the next release.

**Comment by SchoolGuy** [comment]
Context: https://askubuntu.com/questions/142816/which-debconf-preseed-option-to-install-grub-automatically-to-mbr

**Comment by SchoolGuy** [comment]
Since I want to get the release out and it is highly likely that this is not a Cobbler Bug, I will relabel and unschedule this. Trying this out means I need to get familiar with preseed since I never touched it before.
